packages/adapter-prisma/lib/adapter-prisma.js

```diff
@@ -0,0 +1,611 @@
+    this._runPrismaCmd('migrate up --experimental');
+  }
+
+  _runPrismaCmd(cmd) {
+    return execSync(`yarn prisma ${cmd} --schema ${this.schemaPath}`, {
+      env: { ...process.env, DATABASE_URL: this._url() },
+      encoding: 'utf-8',
+    });
+  }
+
+  disconnect() {
+    this.prisma.$disconnect();
+    // Everything below here is being cleaned up in an attempt to help out the garbage collector
+    // delete this.prisma;
+    // Object.values(this.listAdapters).forEach(listAdapter => {
+    //   delete listAdapter.prisma;
+    // });
+    // delete require.cache[require.resolve(this.clientPath)];
+  }
+
+  getDefaultPrimaryKeyConfig() {
+    // Required here due to circular refs
+    const { AutoIncrement } = require('@keystonejs/fields-auto-increment');
+    return AutoIncrement.primaryKeyDefaults[this.name].getConfig();
+  }
+
+  async checkDatabaseVersion() {
+    // FIXME: Decide what/how we want to check things here
+  }
+}
+
+class PrismaListAdapter extends BaseListAdapter {
+  constructor(key, parentAdapter) {
+    super(...arguments);
+    this.getListAdapterByKey = parentAdapter.getListAdapterByKey.bind(parentAdapter);
+  }
+
+  _postConnect({ rels, prisma }) {
+    // https://www.prisma.io/docs/reference/tools-and-interfaces/prisma-schema/models#queries-crud
+    // "By default the name of the property is the lowercase form of the model name,
+    // e.g. user for a User model or post for a Post model."
+    this.model = prisma[this.key.slice(0, 1).toLowerCase() + this.key.slice(1)];
+    this.fieldAdapters.forEach(fieldAdapter => {
+      fieldAdapter.rel = rels.find(
+        ({ left, right }) =>
+          left.adapter === fieldAdapter || (right && right.adapter === fieldAdapter)
+      );
+    });
+  }
+
+  ////////// Mutations //////////
+  _include() {
+    // We don't have a "real key" (i.e. a column in the table) if:
+    //  * We're a N:N
+    //  * We're the right hand side of a 1:1
+    //  * We're the 1 side of a 1:N or N:1 (e.g we are the one with config: many)
+    const include = defaultObj(
+      this.fieldAdapters
+        .filter(({ isRelationship }) => isRelationship)
+        .filter(a => a.config.many || (a.rel.cardinality === '1:1' && a.rel.right.adapter === a))
+        .map(a => a.path),
+      { select: { id: true } }
+    );
+    return Object.keys(include).length > 0 ? include : undefined;
+  }
+
+  async _create(_data) {
+    return this.model.create({
+      data: mapKeys(_data, (value, path) =>
+        this.fieldAdaptersByPath[path] && this.fieldAdaptersByPath[path].isRelationship
+          ? {
+              connect: Array.isArray(value)
+                ? value.map(x => ({ id: Number(x) }))
+                : { id: Number(value) },
+            }
+          : value
+      ),
+      include: this._include(),
+    });
+  }
+
+  async _update(id, _data) {
+    const include = this._include();
+    const existingItem = await this.model.findOne({ where: { id: Number(id) }, include });
```

Comment:
This wouldn't work for string ids right?

(this comment is in reference to all the Number calls in this file, not just this one)

Reply:
Yep, good catch, I need to loop back and remove the need for all these `Number` conversions, because they shouldn't be happening in this part of the adapter.

Reply:
For now I'm going to say that we only support `AutoIncrement` as the field type for the `id` field. We need to introduce better testing for the other cases before we decide how best to clean this stuff up.